### ISSUE TITLE
Out of memory error

### ISSUE THREAD

**Issue by Gootjes** [gamlj/gamlj#180]
Hi there,

I am running gamlj from a normal R session (instead of from jamovi).

Since R 4.2, memory allocation in R for Windows has changed. Limits have been removed.
This means the following line currently leads to a crash of my laptop:
```r
Z %*% Sigma %*% t(Z)
```
https://github.com/gamlj/gamlj/blob/master/R/lmer.rsquared.R#L40

I traced it to that multiplication statement because in my case, `Z` is of length 49024 (dataset is around 51404 observations, minus some missing values I guess). A matrix multiplication of this magnitude led to this error in 4.1.2:
`Error: cannot allocate vector of size 17.9 Gb` 

However, in 4.3.0 and 4.2.2, no such error occurs, the laptop simply tries to execute the code even though it can't, choking the OS (and ending in the OS killing the whole process).
I see that you built in a tryCatch around the rsquare calculation, producing NA and an error in case the code fails:
```r
r2<-try(r.squared(model),silent = TRUE)
``` 
https://github.com/gamlj/gamlj/blob/master/R/gamljmixed.b.R#L326

But since 4.2 / 4.3, the code does not fail this way anymore.

I tried figuring out a solution to this, but the way to set a memory limit for the R process has disappeared on Windows?
https://community.rstudio.com/t/how-to-increase-the-memory-allocation-after-r-version-4-2/148935

So in other words, my OS (Windows 10) is not doing a great job of setting a limit on R (since technically the limit is 8Tb), and the manual way of setting a limit on R has disappeared.

Maybe it is a solution to not even try the rsquared method for vectors longer than X?
Or, maybe use the package `memuse::Sys.meminfo()` ?

reprex:

In R windows 4.1 64-bit try this code that leads to a memory error:
```r
arr <- array(1, 49024)
arr %*% t(arr)
```

In R windows 4.3 64-bit try this code that leads to no memory error:
```r
arr <- array(1, 49024)
arr %*% t(arr)
```

**Comment by mcfanda** [comment]
Does ``memory.limit()`` work?

**Comment by Gootjes** [comment]
No they also deprecated `memory.limit()` since R 4.2

Hmmm, maybe this is really the issue of R and Windows

**Comment by Gootjes** [comment]
Nice. Looking forward to the next version! And I hope I don't to look back to my old gamlj syntax.... :)

**Comment by Gootjes** [comment]
Okay, so I ran a bit of testcode, and also cross-validated it with `memuse::howbig(1000, 1000)`, although that function returns slightly different values. My data is nicely fitting the following function:
Every numeric is 8 bytes.
```r
memory_usage <- function(n) n*n*8 # for a n by n matrix of type numeric)
```

With the use of the memuse package, we can do the following:
```r
max_dimensions_of_matrix <- memuse::howmany(memuse::Sys.meminfo()$totalram)

```

**Comment by mcfanda** [comment]
that nice, I'll work on this.
